src/legacy/core_plugins/console/public/src/components/settings_modal.tsx

```diff
@@ -0,0 +1,263 @@
+  const [wrapMode, setWrapMode] = useState(props.settings.wrapMode);
+  const [fields, setFields] = useState(props.settings.autocomplete.fields);
+  const [indices, setIndices] = useState(props.settings.autocomplete.indices);
+  const [templates, setTemplates] = useState(props.settings.autocomplete.templates);
+  const [polling, setPolling] = useState(props.settings.polling);
+  const [tripleQuotes, setTripleQuotes] = useState(props.settings.tripleQuotes);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    return () => {
+      // TODO
+    };
+  }, []);
+
+  function saveSettings() {
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    props.onSaveSettings({
+      fontSize,
+      wrapMode,
+      autocomplete: {
+        fields,
+        indices,
+        templates,
+      },
+      polling,
+      tripleQuotes,
+    });
+
+    setIsLoading(false);
+  }
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal
+        data-test-subj="devToolsSettingsModal"
+        className="devToolsSettingsModal"
+        onClose={props.onClose}
+      >
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage id="console.settingsPage.pageTitle" defaultMessage="Settings" />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody className="euiText euiText--medium euiText--constrainedWidth">
+          <EuiFlexGrid columns={1}>
+            <EuiFlexItem>
+              <EuiFlexItem>
+                <EuiText>
+                  <p>
+                    <FormattedMessage
+                      id="console.settingsPage.fontSizeLabel"
+                      defaultMessage="Font Size"
+                    />
+                  </p>
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiFieldNumber
+                  autoFocus
+                  data-test-subj="fontSizeInput"
+                  value={fontSize}
+                  onChange={e => {
+                    setFontSize((e.target.value as unknown) as number);
+                  }}
+                />
+              </EuiFlexItem>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiSwitch
+                checked={wrapMode}
+                data-test-subj="settingsWrapLines"
+                id="wrapLines"
+                label={
+                  <FormattedMessage
+                    defaultMessage="Wrap Long Lines"
+                    id="console.settingsPage.wrapLongLinesLabelText"
+                  />
+                }
+                onChange={e => setWrapMode(e.target.checked)}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <h4>
+                <FormattedMessage
+                  id="console.settingsPage.jsonSyntaxLabel"
+                  defaultMessage="JSON syntax"
+                />
+              </h4>
```

Comment:
Wrap all these h4's in a `<EuiTitle size="xs">` for styling

Reply:
Used formrow labels instead